docs/src/main/tut/docs/stack/README.md

```diff
@@ -0,0 +1,260 @@
+
+We will represent these capabilities using two algebras: `CustomerPersistence` and `StockPersistence`.
+
+```tut:book
+object algebras {
+  @free trait CustomerPersistence[F[_]] {
+    def getCustomer(id: CustomerId): FreeS.Par[F, Option[Customer]]
+  }
+
+  @free trait StockPersistence[F[_]] {
+    def checkQuantityAvailable(cultivar: String): FreeS.Par[F, Int]
+    def registerOrder(order: Order): FreeS.Par[F, Unit]
+  }
+}
+```
+
+These two algebras will be combined together in a `Persistence` module, that in its turn will be a part of our `App` module, that additionally includes the
+capabilities to fail, to read from our `Config` configuration and to accumulate some logging messages using respectively the `error`, the `reader` and  the `writer` effect.
+
+```tut:book
+import freestyle.effects.error._
+import freestyle.effects.{reader, writer}
+
+object modules {
+  val rd = reader[Config]
+  val wr = writer[Vector[String]]
+
+  @module trait Persistence[F[_]] {
+    val customer: algebras.CustomerPersistence[F]
+    val stock: algebras.StockPersistence[F]
+  }
+
+  @module trait App[F[_]] {
+    val persistence: Persistence[F]
+
+    val errorM: ErrorM[F]
+    val readerM: rd.ReaderM[F]
+    val writerM: wr.WriterM[F]
+  }
+}
+```
+
+To validate the order we check if the number of crates that is ordered bigger is than zero and that the requested apple cultivar is selled by the company.
+
+For the second part we use the `reader` effect algebra to read the set of apple cultivars from our `Config`.
+
+```tut:book
+import cats.implicits._
+
+import modules._
+import cats.data.{NonEmptyList, ValidatedNel}
+
+def validateOrder[F[_]](order: Order, customer: Customer)(implicit app: App[F]): FreeS.Par[F, ValidatedNel[String, Unit]] =
+  app.readerM.reader { config =>
+    val v = ().validNel[String]
+    v.ensure(NonEmptyList.of(
+      "Number of crates ordered should be bigger than zero"))(
+      _ => order.crates > 0) |+|
+    v.ensure(NonEmptyList.of(
+      "Apple cultivar is not available"))(
+      _ => config.cultivars.contains(order.cultivar.toLowerCase))
+  }
+```
+
+
+We will create a couple of domain specific exception case classes.
+
+```tut:book
+sealed abstract class AppleException(val message: String) extends Exception(message)
+case class CustomerNotFound(id: CustomerId)               extends AppleException(s"Customer $id can not be found")
+case class QuantityNotAvailable(error: String)            extends AppleException(error)
+case class ValidationFailed(errors: NonEmptyList[String]) extends AppleException(errors.intercalate("\n"))
+```
+
+We can use the `validateOrder` method in combination with our persistence algebras and the `error` and `writer` effect algebras, to create the `processOrder` method tying everything together.
+
+```tut:book
+def processOrder[F[_]](order: Order)(implicit app: App[F]): FreeS[F, String] =
+  for {
+    customerOpt <- app.persistence.customer.getCustomer(order.customerId)
```

Comment:
instead of having long lines of `app.persistence.customer.getCustomer` can we have instead those imported so the for comprehension has as little noise as possible?